### ISSUE TITLE
Idefics - AttentionMasks wrongly set with padding='longest'

### ISSUE THREAD

**Issue by VictorSanh** [huggingface/transformers#28591]
### System Info

transformers==4.36.2

### Reproduction

Reported by https://huggingface.co/VishnuSuganth
https://huggingface.co/HuggingFaceM4/idefics-9b-instruct/discussions/11

**Comment by amyeroberts** [comment]
Cc @ArthurZucker @younesbelkada

**Comment by ArthurZucker** [comment]
Might be a tokenization issue will have a look

**Comment by byi8220** [comment]
Is anyone working on this issue? If not, would it be something a new contributor could look at?

I think the issue may be how `unpadded_seq_len` is calculated here: https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics/processing_idefics.py#L347-L354

Dumping the parameters inside this loop we can see a mismatch between the case of `padding=False` and `padding='longest'` for the smaller input in their repro:
```
Calling processor() with padding=False
--------------------------------------------------
Param Values:
text=[1, 4911, 29901, 1724, 338, 297, 445, 1967, 29973, 32000, 32001, 32000]
decoded_text=<s> User: What is in this image?<fake_token_around_image><image><fake_token_around_image>
unpadded_seq_len=12
start=8
```
```
Calling processor() with padding='longest'
--------------------------------------------------
Param Values:
text=[0, 0, 0, 0, 0, 0, 0, 0, 1, 4911, 29901, 1724, 338, 297, 445, 1967, 29973, 32000, 32001, 32000]
decoded_text=<unk><unk><unk><unk><unk><unk><unk><unk><s> User: What is in this image?<fake_token_around_image><image><fake_token_around_image>
unpadded_seq_len=20
start=0
```

If using `<unk>` as the padding token (https://huggingface.co/HuggingFaceM4/idefics-9b-instruct/blob/main/tokenizer_config.json#L64) is deliberate, could this issue be solved by simply omitting the padding token from the calculation of `unpadded_seq_len`, or is this problem more complex?

**Comment by byi8220** [comment]
After looking closer at `IdeficsProcessor`, I have a (possibly overconfident) idea of what's going on.

IIUC, it seems like this code is currently **always** hand-performing an additional padding pass after the tokenizer performs it's padding, according to this code: https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics/processing_idefics.py#L348-L354

And since the output tensors are stacked together [in this code](https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics/processing_idefics.py#L372-L374), they all have to be of the same size, i.e. padded or otherwise normalized.

In this case, I am not sure if `padding` should default to `False`, and instead should default to `longest`.

^ Tried my hand at a fix in the pr above #29449

**Comment by ArthurZucker** [comment]
Thanks I'll review in a bit!
